src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3034,7 +3034,9 @@ class CgroupUtils(object):
                        % (str(filename), msg.strip()))
             if filename is None:
                 return
-            with open(filename, 'a') as desc:
+            # sticky bit requires leaving out os.O_CREAT
+            fd = os.open(filename, os.O_APPEND|os.O_WRONLY)
+            with os.fdopen(fd, 'a') as desc:
```

Comment:
Excuse my ignorance here.  Why can't we just modify the os.open().  You are already opening it O_APPEND.  Also, if we can't O_CREAT, won't that mean we will never create the file?

Reply:
Hi @bhroam ! The function `os.open()` is not the built-in function `open()`. The `open()` returns a python file object but we can not omit O_CREAT with just `open()`. That is why we use `os.open()` here. The function `os.open()` returns just a number - the file descriptor, but we need the *file object* to modify. The file object is retrieved from the fd by `os.fdopen()`. The function `os.fdopen()` is alias for the built-in function `open()`. That is why we need the two functions here.

Yes, we will never create the file here. In case of such an exception - the file does not exist, we are probably in an interactive job, which means no stderr file. We don't need/want to create here. The exception is passed and it is OK.